plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/HudiDirectoryLister.java

```diff
@@ -27,7 +27,9 @@
 {
     List<HudiPartitionInfo> getPartitionsToScan();
 
-    List<FileStatus> listStatus(HudiPartitionInfo partitionInfo);
+    String getMaxCommitTime();
```

Comment:
This looks like a timeline API, doesn't really need to be part of the interface definition. Since it is use in `listFileSlice` method below, let's remove this method and get the timeline inside `listFileSlice` implementation.